From 6f43145fa4f7d5d9d7b260823581d7cf5431122f Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Sun, 24 Aug 2025 13:00:16 +0200
Subject: [PATCH 093/115] mfd: ti-lmu: Remove cells lacking upstreamed drivers

This removes lm3633 and lm3695 as they lack any drivers now.

Signed-off-by: Dinolek <git@dinolek.me>
---
 drivers/mfd/ti-lmu.c                | 42 --------------
 include/linux/mfd/ti-lmu-register.h | 89 -----------------------------
 include/linux/mfd/ti-lmu.h          |  2 -
 3 files changed, 133 deletions(-)

diff --git a/drivers/mfd/ti-lmu.c b/drivers/mfd/ti-lmu.c
index aa9df0d31..2ef1c6ae9 100644
--- a/drivers/mfd/ti-lmu.c
+++ b/drivers/mfd/ti-lmu.c
@@ -70,48 +70,12 @@ static const struct mfd_cell lm3631_devices[] = {
 	LM363X_REGULATOR(LM3631_LDO_OREF),
 	LM363X_REGULATOR(LM3631_LDO_POS),
 	LM363X_REGULATOR(LM3631_LDO_NEG),
-	{
-		.name          = "ti-lmu-backlight",
-		.id            = LM3631,
-		.of_compatible = "ti,lm3631-backlight",
-	},
 };
 
 static const struct mfd_cell lm3632_devices[] = {
 	LM363X_REGULATOR(LM3632_BOOST),
 	LM363X_REGULATOR(LM3632_LDO_POS),
 	LM363X_REGULATOR(LM3632_LDO_NEG),
-	{
-		.name          = "ti-lmu-backlight",
-		.id            = LM3632,
-		.of_compatible = "ti,lm3632-backlight",
-	},
-};
-
-static const struct mfd_cell lm3633_devices[] = {
-	{
-		.name          = "ti-lmu-backlight",
-		.id            = LM3633,
-		.of_compatible = "ti,lm3633-backlight",
-	},
-	{
-		.name          = "lm3633-leds",
-		.of_compatible = "ti,lm3633-leds",
-	},
-	/* Monitoring driver for open/short circuit detection */
-	{
-		.name          = "ti-lmu-fault-monitor",
-		.id            = LM3633,
-		.of_compatible = "ti,lm3633-fault-monitor",
-	},
-};
-
-static const struct mfd_cell lm3695_devices[] = {
-	{
-		.name          = "ti-lmu-backlight",
-		.id            = LM3695,
-		.of_compatible = "ti,lm3695-backlight",
-	},
 };
 
 static const struct mfd_cell lm36274_devices[] = {
@@ -135,8 +99,6 @@ static const struct ti_lmu_data chip##_data =	\
 
 TI_LMU_DATA(lm3631, LM3631_MAX_REG);
 TI_LMU_DATA(lm3632, LM3632_MAX_REG);
-TI_LMU_DATA(lm3633, LM3633_MAX_REG);
-TI_LMU_DATA(lm3695, LM3695_MAX_REG);
 TI_LMU_DATA(lm36274, LM36274_MAX_REG);
 
 static int ti_lmu_probe(struct i2c_client *cl)
@@ -205,8 +167,6 @@ static int ti_lmu_probe(struct i2c_client *cl)
 static const struct of_device_id ti_lmu_of_match[] = {
 	{ .compatible = "ti,lm3631", .data = &lm3631_data },
 	{ .compatible = "ti,lm3632", .data = &lm3632_data },
-	{ .compatible = "ti,lm3633", .data = &lm3633_data },
-	{ .compatible = "ti,lm3695", .data = &lm3695_data },
 	{ .compatible = "ti,lm36274", .data = &lm36274_data },
 	{ }
 };
@@ -215,8 +175,6 @@ MODULE_DEVICE_TABLE(of, ti_lmu_of_match);
 static const struct i2c_device_id ti_lmu_ids[] = {
 	{ "lm3631", LM3631 },
 	{ "lm3632", LM3632 },
-	{ "lm3633", LM3633 },
-	{ "lm3695", LM3695 },
 	{ "lm36274", LM36274 },
 	{ }
 };
diff --git a/include/linux/mfd/ti-lmu-register.h b/include/linux/mfd/ti-lmu-register.h
index 03f31f3ae..182064a6c 100644
--- a/include/linux/mfd/ti-lmu-register.h
+++ b/include/linux/mfd/ti-lmu-register.h
@@ -98,95 +98,6 @@
 
 #define LM3632_MAX_REG				0x10
 
-/* LM3633 */
-#define LM3633_REG_HVLED_OUTPUT_CFG		0x10
-#define LM3633_HVLED1_CFG_MASK			BIT(0)
-#define LM3633_HVLED2_CFG_MASK			BIT(1)
-#define LM3633_HVLED3_CFG_MASK			BIT(2)
-#define LM3633_HVLED1_CFG_SHIFT			0
-#define LM3633_HVLED2_CFG_SHIFT			1
-#define LM3633_HVLED3_CFG_SHIFT			2
-
-#define LM3633_REG_BANK_SEL			0x11
-
-#define LM3633_REG_BL0_RAMP			0x12
-#define LM3633_REG_BL1_RAMP			0x13
-#define LM3633_BL_RAMPUP_MASK			0xF0
-#define LM3633_BL_RAMPUP_SHIFT			4
-#define LM3633_BL_RAMPDN_MASK			0x0F
-#define LM3633_BL_RAMPDN_SHIFT			0
-
-#define LM3633_REG_BL_RAMP_CONF			0x1B
-#define LM3633_BL_RAMP_MASK			0x0F
-#define LM3633_BL_RAMP_EACH			0x05
-
-#define LM3633_REG_PTN0_RAMP			0x1C
-#define LM3633_REG_PTN1_RAMP			0x1D
-#define LM3633_PTN_RAMPUP_MASK			0x70
-#define LM3633_PTN_RAMPUP_SHIFT			4
-#define LM3633_PTN_RAMPDN_MASK			0x07
-#define LM3633_PTN_RAMPDN_SHIFT			0
-
-#define LM3633_REG_LED_MAPPING_MODE		0x1F
-#define LM3633_LED_EXPONENTIAL			BIT(1)
-
-#define LM3633_REG_IMAX_HVLED_A			0x20
-#define LM3633_REG_IMAX_HVLED_B			0x21
-#define LM3633_REG_IMAX_LVLED_BASE		0x22
-
-#define LM3633_REG_BL_FEEDBACK_ENABLE		0x28
-
-#define LM3633_REG_ENABLE			0x2B
-#define LM3633_LED_BANK_OFFSET			2
-
-#define LM3633_REG_PATTERN			0x2C
-
-#define LM3633_REG_BOOST_CFG			0x2D
-#define LM3633_OVP_MASK				(BIT(1) | BIT(2))
-#define LM3633_OVP_40V				0x6
-
-#define LM3633_REG_PWM_CFG			0x2F
-#define LM3633_PWM_A_MASK			BIT(0)
-#define LM3633_PWM_B_MASK			BIT(1)
-
-#define LM3633_REG_BRT_HVLED_A_LSB		0x40
-#define LM3633_REG_BRT_HVLED_A_MSB		0x41
-#define LM3633_REG_BRT_HVLED_B_LSB		0x42
-#define LM3633_REG_BRT_HVLED_B_MSB		0x43
-
-#define LM3633_REG_BRT_LVLED_BASE		0x44
-
-#define LM3633_REG_PTN_DELAY			0x50
-
-#define LM3633_REG_PTN_LOWTIME			0x51
-
-#define LM3633_REG_PTN_HIGHTIME			0x52
-
-#define LM3633_REG_PTN_LOWBRT			0x53
-
-#define LM3633_REG_PTN_HIGHBRT			LM3633_REG_BRT_LVLED_BASE
-
-#define LM3633_REG_BL_OPEN_FAULT_STATUS		0xB0
-
-#define LM3633_REG_BL_SHORT_FAULT_STATUS	0xB2
-
-#define LM3633_REG_MONITOR_ENABLE		0xB4
-
-#define LM3633_MAX_REG				0xB4
-
-/* LM3695 */
-#define LM3695_REG_GP				0x10
-#define LM3695_BL_CHANNEL_MASK			BIT(3)
-#define LM3695_BL_DUAL_CHANNEL			0
-#define LM3695_BL_SINGLE_CHANNEL			BIT(3)
-#define LM3695_BRT_RW_MASK			BIT(2)
-#define LM3695_BL_EN_MASK			BIT(0)
-
-#define LM3695_REG_BRT_LSB			0x13
-#define LM3695_REG_BRT_MSB			0x14
-
-#define LM3695_MAX_REG				0x14
-
 /* LM36274 */
 #define LM36274_REG_REV				0x01
 #define LM36274_REG_BL_CFG_1			0x02
diff --git a/include/linux/mfd/ti-lmu.h b/include/linux/mfd/ti-lmu.h
index 0bc0e8199..cf83d35a1 100644
--- a/include/linux/mfd/ti-lmu.h
+++ b/include/linux/mfd/ti-lmu.h
@@ -21,8 +21,6 @@
 enum ti_lmu_id {
 	LM3631,
 	LM3632,
-	LM3633,
-	LM3695,
 	LM36274,
 	LMU_MAX_ID,
 };
-- 
2.56.0

